### PR TITLE
DEV: remove slightly less trivial `{{action}}` usages

### PR DESCRIPTION
**Pull request by chancancode**
As promised, this is the follow-up to #24278 that is slightly less trivial.

* Some were "trivial" usages that were missed in the previous PR because the same file that had at least one other non-trivial usage.
* These involve extra arguments or inheritance but I have checked that they seem correct.

Same as the other PR:

> To aid review, each file is converted in an individual commit, and the matching method is temporary annotated with `@__action__` instead of the normal `@action`. This forces a git diff when it is already annotated as `@action`.
> 
> In the final commit, I change everything back, so as long as we merge with squash, it should be all good.

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/components/chat/modal/edit-channel-name.hbs [hunk 1]
```diff
@@ -34,7 +34,7 @@
         name="channel-slug"
         class="chat-channel-edit-name-slug-modal__slug-input"
         placeholder={{this.autoGeneratedSlug}}
-        {{on "input" (action "onChangeChatChannelSlug" value="target.value")}}
+        {{on "input" this.onChangeChatChannelSlug}}
```
Comment (chancancode) [inline plugins/chat/assets/javascripts/discourse/components/chat/modal/edit-channel-name.hbs:37]: So this is a little strange, the argument was never actually used in `onChangeChatChannelSlug` – it actually references `this.editedName` and doesn't do anything with `this.editedSlug`. That may or may not be correct but in any case this argument can be removed

File: plugins/chat/assets/javascripts/discourse/components/chat/modal/edit-channel-name.js [hunk 2]
```diff
@@ -48,9 +48,9 @@ export default class ChatModalEditChannelName extends Component {
   }
 
   @action
-  onChangeChatChannelName(title) {
+  onChangeChatChannelName(event) {
     this.flash = null;
-    this.#debouncedGenerateSlug(title);
+    this.#debouncedGenerateSlug(event?.target?.value);
```
Comment (chancancode) [inline plugins/chat/assets/javascripts/discourse/components/chat/modal/edit-channel-name.js:53]: I did the most conservative `?.` here but that is probably not actually necessary, the function is not called anywhere else so it should always have `event.target.value`. Alternatively we could probably just use `this.editedName` here, but I wasn't sure if there are going to be subtle timing differences so I left it, but someone can verify that and do the refactor if desired.

**Comment by chancancode** [comment]
Hi @jjaffeux 👋🏼 mind giving this a review? If we merge this before I re-run the gjs codemod, I think we can maybe convert an extra component 🙂

File: plugins/chat/assets/javascripts/discourse/components/chat/modal/edit-channel-name.js (hunk 2 above)
Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/chat/modal/edit-channel-name.js:53]: Yes fine with this for now thanks

**Review by jjaffeux — APPROVED** [review]

**Comment by jjaffeux** [comment]
Looks good to me @chancancode, will merge if that's ok with you?

**Pull request merged**